agent/doctor/doctor.go

```diff
@@ -35,72 +30,92 @@ type Doctor struct {
 	lock                 sync.RWMutex
 	cluster              string
 	containerInstanceArn string
+	statusReported       bool
 }
 
 func NewDoctor(healthchecks []Healthcheck, cluster string, containerInstanceArn string) (*Doctor, error) {
 	newDoctor := &Doctor{
 		healthchecks:         []Healthcheck{},
 		cluster:              cluster,
 		containerInstanceArn: containerInstanceArn,
+		statusReported:       false,
 	}
 	for _, hc := range healthchecks {
 		newDoctor.AddHealthcheck(hc)
 	}
 	return newDoctor, nil
 }
 
+// GetCluster returns the cluster that was provided to the doctor while
+// being initialized
+func (doc *Doctor) GetCluster() string {
+	doc.lock.RLock()
+	defer doc.lock.RUnlock()
+
+	return doc.cluster
+}
+
+// GetContainerInstanceArn returns the container instance arn that was
+// provided to the doctor while being initialized
+func (doc *Doctor) GetContainerInstanceArn() string {
+	doc.lock.RLock()
+	defer doc.lock.RUnlock()
+
+	return doc.containerInstanceArn
+}
+
+// SetStatusReported tells the doctor that we have already reported the
+// current status of the healthchecks to the backend
+func (doc *Doctor) SetStatusReported(statusReported bool) {
+	doc.lock.Lock()
+	defer doc.lock.Unlock()
+
+	doc.statusReported = statusReported
+}
+
+// HasStatusBeenReported returns whether we have already sent the current
+// state of the healthchecks to the backend or not
+func (doc *Doctor) HasStatusBeenReported() bool {
```

Comment:
`GetStatusReported` instead of `HasStatusBeenReported` ?

Reply:
I think `HasStatusBeenReported` gives more information about what the method is and what it returns. You expect a boolean value to come out of a method called `HasStatusBeenReported` vs `GetStatusReported` could potentially return an enum/string/int value as well as a boolean value, it is ambiguous. 

The [Golang source](https://golang.org/src/go/types/predicates.go) itself uses `is` prefixed boolean methods and [Docker](https://github.com/moby/moby/blob/master/cli/cobra.go#L45-L51) uses `has` prefixes.